include/caffe/util/math_functions.hpp

```diff
@@ -12,6 +12,18 @@
 
 namespace caffe {
 
+template<typename Dtype>
+Dtype caffe_ipow(Dtype base, int exp) {
```

Comment:
I did a small external benchmark. 3sec vs 7sec for caffe_ipow vs std::pow is a difference in speed (multiple iterations). This binary exponenation is common in many (crypto.) programs for several reasons.